eth/vm/forks/homestead/__init__.py

```diff
@@ -18,7 +15,7 @@
 
 class MetaHomesteadVM(FrontierVM):
     support_dao_fork = True
-    dao_fork_block_number = DAO_FORK_MAINNET_BLOCK
+    dao_fork_block_number = 0
```

Comment:
cc @carver thoughts on how we should proceed here? I'm thinking of changing both of this to default to `None` and to add a runtime check that blows up if the value hasn't been explicitely set.

Reply:
How about:

```py
@abstractmethod
@property
def doa_fork_block_number(self):
  pass
```

Reply:
This didn't work well due to use using inheritance for chains.  I went with a `classmethod` implementation.  take a look and tell me what you think.